### ISSUE TITLE
google_cloud_run_domain_mapping require a specific annotation

### ISSUE THREAD

**Issue by steren** [hashicorp/terraform-provider-google#7736]
Domain Mappings for Cloud Run are a Beta feature. The Cloud Run API enforces the presence of a specific annotation when using any Beta feature:
```
    annotations = {
       "run.googleapis.com/launch-stage" = "BETA"
    }
```

I suggest Terraform to auto add this annotation when creating a `google_cloud_run_domain_mapping` resource.

Or alternatively, to better document this, notably by updating the [example on this page](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloud_run_domain_mapping#example-usage---cloud-run-domain-mapping-basic)

**Comment by rileykarson** [comment]
Tagged as bug because not having this annotation breaks the resource right now, after a breaking change in the API.

Note that we got this resource's lifecycle stage wrong. It should have only been added to the `google-beta` provider. It had a `v1` API, and the REST reference had no indicator it was non-GA: https://cloud.google.com/run/docs/reference/rest/v1/namespaces.domainmappings

We probably want to add an infobox indicating that it's beta, but not actually remove it from the `google` provider since that removal would be a breaking change and the resource will GA eventually.

**Comment by slevenick** [comment]
@steren I believe this annotation will not be enforced anymore. Will it be enforced again in the future, or for other annotations?

**Comment by steren** [comment]
Because the feature is in beta, is preferred for client to set the
annotation.

**Comment by tek08** [comment]
I think there's a still a bug here, though maybe a different bug...  With this annotation set, terraform insists on replacing the mapping even if nothing has changed.  I've taken to commenting out the annotation once the mapping is created initially, such as to not cause teardowns of existing infra.

Here's the reasoning terraform insists on the replacement, which lends me to think that #7583 (which was reportedly just fixed and released in 3.47.0) might still be germane:

```
      ~ metadata {
          ~ annotations      = { # forces replacement
                "run.googleapis.com/launch-stage"  = "BETA"
              - "serving.knative.dev/creator"      = "terraform-service-account@<redacted>.iam.gserviceaccount.com" -> null
              - "serving.knative.dev/lastModifier" = "terraform-service-account@<redacted>.iam.gserviceaccount.com" -> null
            }
           ...
```

Happy to provide extra info if it helps triage further.  Thanks y'all.

**Comment by slevenick** [comment]
Hey @tek08 that should be fixed with the next release via: https://github.com/GoogleCloudPlatform/magic-modules/pull/4209
